pkg/cmd/task/start_test.go

```diff
@@ -94,11 +94,34 @@ func Test_start_has_task_arg(t *testing.T) {
 	c := Command(&test.Params{})
 
 	_, err := test.ExecuteCommand(c, "start", "-n", "ns")
-
 	if err == nil {
 		t.Error("Expecting an error but it's empty")
 	}
-	test.AssertOutput(t, "missing task name", err.Error())
+	test.AssertOutput(t, "Either a task name or a --filename parameter must be supplied", err.Error())
+}
+
+func Test_start_has_task_filename(t *testing.T) {
+	ns := []*corev1.Namespace{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "ns",
+			},
+		},
+	}
+	cs, _ := test.SeedTestData(t, pipelinetest.Data{Namespaces: ns})
+	c := Command(&test.Params{Tekton: cs.Pipeline, Kube: cs.Kube})
+
+	oldParseTask := parseTask
+	defer func() {
+		parseTask = oldParseTask
+	}()
+	parseTask = func(p string) (*v1alpha1.Task, error) {
```

Comment:
This doesn't appear to cover `parseTask` in `startTask`.

Reply:
Ah, I see. It's because of the reassignment. Wondering if there's a better testing approach that can be done here, but I understand the reasoning.

Reply:
The only other mechanism I can think of is to "inject" it through the startOptions struct into startTask.

Any thoughts on which you'd prefer?

Reply:
I am comfortable keeping this as is for now, and we can iterate on it before next release. But we can also see if other folks have thoughts.